cinema/views.py

```diff
@@ -48,6 +82,32 @@
     queryset = MovieSession.objects.all()
     serializer_class = MovieSessionSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset.prefetch_related("movie", "cinema_hall")
+
+        if self.action == "list":
+            queryset = queryset.annotate(
+                tickets_available=(
+                    F("cinema_hall__rows") * F("cinema_hall__seats_in_row")
+                    - Count("tickets")
+                )
+            )
+
+            filters = {}
+
+            date = self.request.query_params.get("date")
+            movie = self.request.query_params.get("movie")
+
+            if date:
+                filters["show_time__date"] = date
```

Comment:
Code Efficiency: You're splitting the date here, but it's not necessary. You can directly filter by the date as it's already in the format needed. The correct way would be: `filters["show_time__date"] = date`.